app/main.py

```diff
@@ -1 +1,36 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self,
+                 name: str,
+                 health: int = 100,
+                 hidden: bool = False) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        self.alive.append(self)
+
+    def __repr__(self) -> str:
+        return f"{{Name: {self.name}," \
+               f" Health: {self.health}," \
+               f" Hidden: {self.hidden}}}"
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+
+    @staticmethod
+    def bite(bitten_animal: "Animal") -> None:
+        if isinstance(bitten_animal, Carnivore):
```

Comment:
to be honest, it's hard to read a bit, you can just write:
`if isinstance(herbivore, Herbivore) and not herbivore.hidden:`
and you can delete animal from the list using `remove`

Reply:
Fixed this.